FTBFS: error: ‘dataset’ was not declared in this scope

Tried to compile the current tip of `master`. Standard `cmake` followed by `make` with default options besides a custom `CMAKE_INSTALL_PREFIX`.

```
Scanning dependencies of target calculate-element-wise
[ 67%] Building CXX object Applications/src/CMakeFiles/calculate-element-wise.dir/calculate-element-wise.cc.o
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc: In function ‘int main(int, char**)’:
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc:414:22: error: ‘dataset’ was not declared in this scope
           } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
                      ^~~~~~~
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc:609:20: error: ‘dataset’ was not declared in this scope
         } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
                    ^~~~~~~
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc:633:20: error: ‘dataset’ was not declared in this scope
         } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
                    ^~~~~~~
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc:657:20: error: ‘dataset’ was not declared in this scope
         } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
                    ^~~~~~~
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc:685:20: error: ‘dataset’ was not declared in this scope
         } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
                    ^~~~~~~
/home/gvaillan/workspace/MIRTK/Applications/src/calculate-element-wise.cc:704:18: error: ‘dataset’ was not declared in this scope
       } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
                  ^~~~~~~
Applications/src/CMakeFiles/calculate-element-wise.dir/build.make:62: recipe for target 'Applications/src/CMakeFiles/calculate-element-wise.dir/calculate-element-wise.cc.o' failed
make[2]: *** [Applications/src/CMakeFiles/calculate-element-wise.dir/calculate-element-wise.cc.o] Error 1
CMakeFiles/Makefile2:2732: recipe for target 'Applications/src/CMakeFiles/calculate-element-wise.dir/all' failed
make[1]: *** [Applications/src/CMakeFiles/calculate-element-wise.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,10 @@ os:
   - linux
   - osx
 
+env:
+  - WITH_VTK=OFF
+  - WITH_VTK=ON
+
 dist: trusty
 sudo: required
 

--- a/Applications/src/calculate-element-wise.cc
+++ b/Applications/src/calculate-element-wise.cc
@@ -411,9 +411,13 @@ int main(int argc, char **argv)
           const char *aname = nullptr;
           if (HAS_ARGUMENT) {
             aname = ARGUMENT;
-          } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
-            aname = fname;
-            fname = input_name;
+          } else {
+            #if MIRTK_Image_WITH_VTK
+              if (dataset && dataset->GetPointData()->HasArray(fname)) {
+                aname = fname;
+                fname = input_name;
+              }
+            #endif
           }
           UniquePtr<Mask> op(new Mask(fname));
           if (aname) {
@@ -606,9 +610,13 @@ int main(int argc, char **argv)
         const char *aname = nullptr;
         if (HAS_ARGUMENT) {
           aname = ARGUMENT;
-        } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
-          aname = fname;
-          fname = input_name;
+        } else {
+          #if MIRTK_Image_WITH_VTK
+            if (dataset && dataset->GetPointData()->HasArray(fname)) {
+              aname = fname;
+              fname = input_name;
+            }
+          #endif
         }
         UniquePtr<Add> op(new Add(fname));
         if (aname) {
@@ -630,9 +638,13 @@ int main(int argc, char **argv)
         const char *aname = nullptr;
         if (HAS_ARGUMENT) {
           aname = ARGUMENT;
-        } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
-          aname = fname;
-          fname = input_name;
+        } else {
+          #if MIRTK_Image_WITH_VTK
+            if (dataset && dataset->GetPointData()->HasArray(fname)) {
+              aname = fname;
+              fname = input_name;
+            }
+          #endif
         }
         UniquePtr<Sub> op(new Sub(fname));
         if (aname) {
@@ -654,9 +666,13 @@ int main(int argc, char **argv)
         const char *aname = nullptr;
         if (HAS_ARGUMENT) {
           aname = ARGUMENT;
-        } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
-          aname = fname;
-          fname = input_name;
+        } else {
+          #if MIRTK_Image_WITH_VTK
+            if (dataset && dataset->GetPointData()->HasArray(fname)) {
+              aname = fname;
+              fname = input_name;
+            }
+          #endif
         }
         UniquePtr<Mul> op(new Mul(fname));
         if (aname) {
@@ -682,9 +698,13 @@ int main(int argc, char **argv)
         const char *aname = nullptr;
         if (HAS_ARGUMENT) {
           aname = ARGUMENT;
-        } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
-          aname = fname;
-          fname = input_name;
+        } else {
+          #if MIRTK_Image_WITH_VTK
+            if (dataset && dataset->GetPointData()->HasArray(fname)) {
+              aname = fname;
+              fname = input_name;
+            }
+          #endif
         }
         UniquePtr<Div> op(new Div(fname));
         if (aname) {
@@ -701,9 +721,13 @@ int main(int argc, char **argv)
       const char *aname = nullptr;
       if (HAS_ARGUMENT) {
         aname = ARGUMENT;
-      } else if (dataset && dataset->GetPointData()->HasArray(fname)) {
-        aname = fname;
-        fname = input_name;
+      } else {
+        #if MIRTK_Image_WITH_VTK
+          if (dataset && dataset->GetPointData()->HasArray(fname)) {
+            aname = fname;
+            fname = input_name;
+          }
+        #endif
       }
       UniquePtr<DivWithZero> op(new DivWithZero(fname));
       if (aname) {

--- a/Scripts/install_depends.sh
+++ b/Scripts/install_depends.sh
@@ -21,17 +21,22 @@ if [ $os = linux ] || [ $os = Linux ]; then
     libboost-random-dev \
     libeigen3-dev \
     libflann-dev \
-    libgtest-dev \
     libnifti-dev \
     libpng12-dev \
     libsuitesparse-dev \
-    libtbb-dev \
-    libvtk6-dev
+    libtbb-dev
 
-  mkdir "$HOME/gtest-build" && cd "$HOME/gtest-build"
-  cmake /usr/src/gtest && make
-  sudo mv -f libgtest.a libgtest_main.a /usr/lib
-  rm -rf "$HOME/gtest-build"
+  if [ $WITH_VTK = ON ]; then
+    sudo apt-get install -y --no-install-recommends libvtk6-dev
+  fi
+
+  if [ $TESTING = ON ]; then
+    sudo apt-get install -y --no-install-recommends libgtest-dev
+    mkdir "$HOME/gtest-build" && cd "$HOME/gtest-build"
+    cmake /usr/src/gtest && make
+    sudo mv -f libgtest.a libgtest_main.a /usr/lib
+    rm -rf "$HOME/gtest-build"
+  fi
 
 # install pre-requisites on OS X
 elif [ $os = osx ] || [ $os = Darwin ]; then
@@ -43,13 +48,18 @@ elif [ $os = osx ] || [ $os = Darwin ]; then
     eigen \
     flann \
     suite-sparse \
-    tbb \
-    vtk
+    tbb
 
-  git clone --depth=1 https://github.com/google/googletest.git "$HOME/gtest-source"
-  mkdir "$HOME/gtest-build" && cd "$HOME/gtest-build"
-  cmake -DCMAKE_CXX_FLAGS=-std=c++11 -DBUILD_GMOCK=OFF -DBUILD_GTEST=ON ../gtest-source && make
-  sudo make install
-  rm -rf "$HOME/gtest-source" "$HOME/gtest-build"
+  if [ $WITH_VTK = ON ]; then
+    brew install vtk
+  fi
+
+  if [ $TESTING = ON ]; then
+    git clone --depth=1 https://github.com/google/googletest.git "$HOME/gtest-source"
+    mkdir "$HOME/gtest-build" && cd "$HOME/gtest-build"
+    cmake -DCMAKE_CXX_FLAGS=-std=c++11 -DBUILD_GMOCK=OFF -DBUILD_GTEST=ON ../gtest-source && make
+    sudo make install
+    rm -rf "$HOME/gtest-source" "$HOME/gtest-build"
+  fi
 
 fi

--- a/Scripts/travis.sh
+++ b/Scripts/travis.sh
@@ -1,13 +1,34 @@
 ## Travis CI script
 set -e
 
+norm_option_value()
+{
+  if [ $1 = on ] || [ $1 = yes ] || [ $1 = YES ] || [ $1 = y ] || [ $1 = Y ] || [ $1 = 1 ]; then
+    echo ON
+  else
+    echo OFF
+  fi
+}
+
+TESTING=`norm_option_value "$TESTING"`
+WITH_VTK=`norm_option_value "$WITH_VTK"`
+
+modules=(Common Numerics Image IO Transformation Registration DrawEM)
+if [ $WITH_VTK = ON ]; then
+  modules=(${modules[@]} PointSet Deformable Mapping)
+fi
+
+cmake_args=
+for module in ${modules[@]}; do
+  cmake_args=(${config[@]} -D MODULE_${module}=ON)
+done
+
 mkdir Build && cd Build
 cmake -D CMAKE_INSTALL_PREFIX=$HOME/local \
       -D CMAKE_BUILD_TYPE=Release \
-      -D BUILD_ALL_MODULES=ON \
       -D BUILD_SHARED_LIBS=ON \
       -D BUILD_APPLICATIONS=ON \
-      -D BUILD_TESTING=${tests} \
+      -D BUILD_TESTING=$TESTING \
       -D BUILD_DOCUMENTATION=OFF \
       -D BUILD_CHANGELOG=OFF \
       -D WITH_ARPACK=ON \
@@ -18,9 +39,10 @@ cmake -D CMAKE_INSTALL_PREFIX=$HOME/local \
       -D WITH_PROFILING=ON \
       -D WITH_TBB=ON \
       -D WITH_UMFPACK=ON \
-      -D WITH_VTK=ON \
+      -D WITH_VTK=$WITH_VTK \
       -D WITH_ZLIB=ON \
+      ${cmake_args[@]} \
       ..
 
 make -j 8
-[ ${tests} = off ] || make test
+[ $TESTING = OFF ] || make test

